### PR TITLE
Fix jsConfigHelper lang/locale mixup

### PR DESCRIPTION
**Pull request by rullzer**
To test:

1. Set your language to US english
2. Set your locale to German
3. Reload

Open your js console and get the value of

* datepickerFormatDate
* dayNames

Before:

* datepickerFormatDate  in German format `dd.MM.yy"
* dayNames also in German

Now:

* datepickerFormatDate  in German format `dd.MM.yy"
* dayNames also in English

### CODE REVIEW

**Review by georgehrke — APPROVED** [review]
Works 👍 

Thx a lot!

File: lib/private/legacy/defaults.php [hunk 1]
```diff
@@ -219,6 +217,10 @@ public function getSlogan() {
 		if ($this->themeExist('getSlogan')) {
 			return $this->theme->getSlogan();
 		} else {
+			if ($this->defaultSlogan === null) {
+				$l10n = \OC::$server->getL10N('lib');
```
Comment (nickvergessen) [inline lib/private/legacy/defaults.php]: You can inject the IFactory?

**Comment by kesselb** [comment]
https://github.com/nextcloud/server/pull/11134 &https://github.com/nextcloud/server/issues/11132

Okay. The factory will figure out the right language somehow. Do we still need `findLanguageFromLocale` then and does everything works as before? I think there must be a reason for this previous change.

File: lib/private/legacy/defaults.php (hunk 1 above)
Comment (rullzer) [reply lib/private/legacy/defaults.php]: I'm afraid to inject anything here in this legacy code. I vote to keep this as is for now so it backports clean and then spend time in cleaning up thise theming madness ;)

**Comment by rullzer** [comment]
> #11134 & #11132
> 
> Okay. The factory will figure out the right language somehow. Do we still need `findLanguageFromLocale` then and does everything works as before? I think there must be a reason for this previous change.

The reason is that the server funciton doesn't allow you to set a locale right now (but differnt issue)
Also the caching doesn't take the locale into account.

So this was 2 bugs.

1. The JSConfigHelper got injected with a L10N instance that was set to the locale as language. Which translated all the days etc.
2. Because of using translations in the contructor this was all initialized to soon (before there is a user set) so we could not fetch the proper locale. Which in turn then used the cached one without the locale etc. Lots of fun.

**Comment by skjnldsv** [comment]
Make sure this still works https://github.com/nextcloud/server/issues/11132 :)

**Comment by georgehrke** [comment]
> Make sure this still works #11132 :)

Should it? 
https://github.com/nextcloud/server/pull/11134 is introducing exactly the unexpected behaviour for me.

If my language is german and my locale is US, i want my dates to be MM/DD/YYYY, my time to be 12hours with am/pm and the first day to be Sunday.
But since my language is german, I want the weekdays to be Montag, Dienstag, Mittwoch ...

**Comment by kesselb** [comment]
image

```
Index: apps/settings/js/settings/personalInfo.js
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- apps/settings/js/settings/personalInfo.js	(revision e71f2220821974aaf2f2b3357c0505e13a541b5a)
+++ apps/settings/js/settings/personalInfo.js	(date 1572083533569)
@@ -372,7 +372,7 @@
 	$('#localeexample-time').text(moment().format('LTS'));
 	$('#localeexample-date').text(moment().format('L'));
 	$('#localeexample-fdow').text(t('settings', 'Week starts on {fdow}',
-		{fdow: moment().weekday(0).format('dddd')}));
+		{fdow: dayNames[firstDay]}));
 
 }, 1000);
```

**Review by kesselb — APPROVED** [review]

**Pull request merged**